girder/cli/build.py

```diff
@@ -63,6 +63,19 @@ def main(dev, mode, watch, watch_plugin, npm, reinstall):
         npmLockFile = os.path.join(staging, 'package-lock.json')
         if os.path.exists(npmLockFile):
             os.unlink(npmLockFile)
+
+        # Remove any lingering node_modules to ensure clean install
+        plugin_dirs = [path.replace('file:', '') for path in _collectPluginDependencies().values()]
+        dirs = [staging, os.path.join(staging, 'src')]
+        dirs.extend(plugin_dirs)
+        node_module_dirs = [os.path.join(d, 'node_modules') for d in dirs]
```

Comment:
These ~5 lines of list manipulation could probably be condensed into 2-3 without losing readability, but it's certainly subjective.